src/vs/workbench/contrib/chat/common/chatService.ts

```diff
@@ -54,7 +54,7 @@ export interface IChatProvider {
 	resolveRequest?(session: IChat, context: any, token: CancellationToken): ProviderResult<IChatRequest>;
 	provideWelcomeMessage?(token: CancellationToken): ProviderResult<(string | IChatReplyFollowup[])[] | undefined>;
 	provideFollowups?(session: IChat, token: CancellationToken): ProviderResult<IChatFollowup[] | undefined>;
-	provideReply(request: IChatRequest, progress: (progress: IChatProgress) => void, token: CancellationToken): ProviderResult<IChatResponse>;
+	provideReply(request: IChatRequest, progress: (progress: IChatProgress) => void | ((progress: { content: string }) => void), token: CancellationToken): ProviderResult<IChatResponse>;
```

Comment:
Could this also just be a progress object that contains a promise that resolves to content? The callbacks back and forth are a bit hard to follow